staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -798,6 +841,61 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 		}
 	}
 
+	// this lives here because all API servers need identities, not just the Kube API server
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIServerIdentity) {
+		const apiServerIdentityHookName = "start-generic-apiserver-identity-lease-controller"
+		if !s.isPostStartHookRegistered(apiServerIdentityHookName) {
+			if err := s.AddPostStartHook(apiServerIdentityHookName, func(context PostStartHookContext) error {
+				config := restclient.CopyConfig(s.APIServerIDConfig)
+				// use protobuf here because we know the kube-apiserver supports it for leases
+				config.ContentConfig.ContentType = "application/vnd.kubernetes.protobuf"
```

Comment:
This should happen in the complete so that anyone using the config gets this.  The usage should simply be directly using the apiserveridconfig in the NewForConfig call.